collector/collector_test.go

```diff
@@ -86,7 +86,79 @@ func TestMetricCollection(t *testing.T) {
 	}
 
 	label := prometheus.Labels{"namespace": "test-namespace"}
-	g, e := reconciler.psCollector.durationScheduled.GetMetricWith(label)
+	g, e := reconciler.prCollector.durationScheduled.GetMetricWith(label)
+	assert.NoError(t, e)
+	assert.NotNil(t, g)
+}
+
+func TestTaskRunCollection(t *testing.T) {
+	objs := []client.Object{}
+	scheme := runtime.NewScheme()
+	_ = v1beta1.AddToScheme(scheme)
+	c := fake.NewClientBuilder().WithScheme(scheme).WithObjects(objs...).Build()
+
+	mockTaskRuns := []*v1beta1.TaskRun{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              "test-taskrun-1",
+				Namespace:         "test-namespace",
+				UID:               types.UID("test-taskrun-1"),
+				CreationTimestamp: metav1.NewTime(time.Now().UTC()),
+			},
+			Status: v1beta1.TaskRunStatus{
+				Status: duckv1.Status{
+					ObservedGeneration: 0,
+					Conditions: duckv1.Conditions{{
+						Type:   "Succeeded",
+						Status: corev1.ConditionTrue,
+					}},
+					Annotations: nil,
+				},
+				TaskRunStatusFields: v1beta1.TaskRunStatusFields{
+					StartTime:      &metav1.Time{Time: time.Now().UTC().Add(5 * time.Second)},
+					CompletionTime: &metav1.Time{Time: time.Now().UTC().Add(10 * time.Second)},
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              "test-taskrun-2",
+				Namespace:         "test-namespace",
+				UID:               types.UID("test-taskrun-2"),
+				CreationTimestamp: metav1.NewTime(time.Now().UTC()),
+			},
+			Status: v1beta1.TaskRunStatus{
+				Status: duckv1.Status{
+					ObservedGeneration: 0,
+					Conditions: duckv1.Conditions{{
+						Type:   "Succeeded",
+						Status: corev1.ConditionTrue,
+					}},
+					Annotations: nil,
+				},
+				TaskRunStatusFields: v1beta1.TaskRunStatusFields{
+					StartTime:      &metav1.Time{Time: time.Now().UTC().Add(5 * time.Second)},
+					CompletionTime: &metav1.Time{Time: time.Now().UTC().Add(10 * time.Second)},
+				},
+			},
+		},
+	}
+	reconciler := &ReconcileTaskRun{client: c, trCollector: NewTaskRunCollector()}
+	ctx := context.TODO()
+	for _, pr := range mockTaskRuns {
+		err := c.Create(context.TODO(), pr)
```

Comment:
How about using `ctx` defined in line [147](https://github.com/openshift-pipelines/pipeline-service-exporter/pull/30/files#diff-66bee09f1f39294b9c301a94f017eacd9bf91ad3182c16ca6e2803252e919df7R147) to be consistent?

Reply:
part of next push